lib/new_relic/agent/instrumentation/opensearch.rb

```diff
@@ -0,0 +1,25 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'opensearch/instrumentation'
+require_relative 'opensearch/chain'
+require_relative 'opensearch/prepend'
+
+DependencyDetection.defer do
+  named :opensearch
+
+  depends_on do
+    defined?(OpenSearch)
```

Comment:
Were there any version constraints that we know of?

Reply:
Good question! The tests passed on Ruby 3.4 as far back as 1.x, so I don't think we have any version constraints. The 2.x and 1.x versions are compatible with Ruby 2.4, so I could add them to the Envfile, even though we're not currently running that on the CI. What do you think?

Reply:
Added a 2.4.x compatible version in: 0f77f6064